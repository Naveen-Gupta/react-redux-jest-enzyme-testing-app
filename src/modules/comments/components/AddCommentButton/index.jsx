import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Button from '@tds/core-button';
import Box from '@tds/core-box';
import DimpleDivider from '@tds/core-dimple-divider';

const AddCommentButton = props => {
	const _redirectToAdd = () => {
		props.history.push('/add');
	};

	return (
		<Fragment>
			<DimpleDivider data-test="dimDivAddCommentBeforeContent"/>
			<Box data-test="boxAddComment">
				<Button data-test="btnAddComment" onClick={_redirectToAdd}>
					Add Comment
				</Button>
			</Box>
			<DimpleDivider data-test="dimDivAddCommentAfterContent"/>
		</Fragment>
	);
};

AddCommentButton.propTypes = {
	history: PropTypes.object,
};

export default AddCommentButton;
