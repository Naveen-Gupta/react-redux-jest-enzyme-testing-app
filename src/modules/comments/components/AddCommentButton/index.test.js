import React from 'react';
import { shallow, mount } from 'enzyme';
import { findByTestAtrr } from '../../../../utils';
import AddCommentButton from './index.jsx';

const setUp = (props = {}) => {
  const component = shallow(<AddCommentButton {...props} />);
  return component;
};

describe('AddCommentButton Component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  test('Should render without errors', () => {
    const wrapper = findByTestAtrr(component, 'boxAddComment');
    expect(wrapper.length).toBe(1);
  });

  test('Should render a button', () => {
    const button = findByTestAtrr(component, 'btnAddComment');
    expect(button.length).toBe(1);
  });

  test('Should render a dimpleDivider before add comment content', () => {
    const dimDivAddCommentBeforeContent = findByTestAtrr(component, 'dimDivAddCommentBeforeContent');
    expect(dimDivAddCommentBeforeContent.length).toBe(1);
  });

  test('Should render a dimpleDivider after add comment content', () => {
    const dimDivAddCommentAfterContent = findByTestAtrr(component, 'dimDivAddCommentAfterContent');
    expect(dimDivAddCommentAfterContent.length).toBe(1);
  });

  test('Should render a button with content "Add Comment"', () => {
    const button = findByTestAtrr(component, 'btnAddComment');
    expect(button.containsMatchingElement('Add Comment')).toBeTruthy();
  });

  test('Checking snapshot', () => {
    const wrap = mount(<AddCommentButton />);
    expect(wrap).toMatchSnapshot();
  });
});
