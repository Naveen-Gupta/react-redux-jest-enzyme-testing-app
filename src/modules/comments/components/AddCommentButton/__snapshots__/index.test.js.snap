// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCommentButton Component works 1`] = `
<AddCommentButton>
  <DimpleDivider
    data-test="dimDivAddCommentBeforeContent"
  >
    <DimpleDivider__StyledDimpleDivider
      data-test="dimDivAddCommentBeforeContent"
    >
      <hr
        className="DimpleDivider__StyledDimpleDivider-quxnvh-0 ciSVbq"
        data-test="dimDivAddCommentBeforeContent"
      />
    </DimpleDivider__StyledDimpleDivider>
  </DimpleDivider>
  <Box
    data-test="boxAddComment"
    inline={false}
    tag="div"
  >
    <styled.div
      data-test="boxAddComment"
      inline={false}
      tag="div"
    >
      <div
        className="sc-AxjAm jooBDv"
        data-test="boxAddComment"
      >
        <Button
          data-test="btnAddComment"
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          <Button__StyledButton
            data-test="btnAddComment"
            onClick={[Function]}
            type="button"
            variant="primary"
          >
            <button
              className="Button__StyledButton-sa7xo3-0 fqeNWL"
              data-test="btnAddComment"
              onClick={[Function]}
              type="button"
            >
              <Button__ButtonTextWrapper>
                <span
                  className="Button__ButtonTextWrapper-sa7xo3-1 ghaQNm"
                >
                  Add Comment
                </span>
              </Button__ButtonTextWrapper>
            </button>
          </Button__StyledButton>
        </Button>
      </div>
    </styled.div>
  </Box>
  <DimpleDivider
    data-test="dimDivAddCommentAfterContent"
  >
    <DimpleDivider__StyledDimpleDivider
      data-test="dimDivAddCommentAfterContent"
    >
      <hr
        className="DimpleDivider__StyledDimpleDivider-quxnvh-0 ciSVbq"
        data-test="dimDivAddCommentAfterContent"
      />
    </DimpleDivider__StyledDimpleDivider>
  </DimpleDivider>
</AddCommentButton>
`;
